.github/workflows/node.yml

```diff
@@ -38,37 +55,65 @@ env:
     CARGO_TERM_COLOR: always
 
 jobs:
-    load-engine-matrix:
-      runs-on: ubuntu-latest
-      outputs:
-        matrix: ${{ steps.load-engine-matrix.outputs.matrix }}
-      steps:
-          - name: Checkout
-            uses: actions/checkout@v4
-
-          - name: Load the engine matrix
-            id: load-engine-matrix
-            shell: bash
-            run: echo "matrix=$(jq -c . < .github/json_matrices/engine-matrix.json)" >> $GITHUB_OUTPUT
+    get-matrices:
+        runs-on: ubuntu-latest
+        outputs:
+            should-run: ${{ steps.get-matrices.outputs.should-run }}
+            engine-matrix-output: ${{ steps.get-matrices.outputs.engine-matrix-output }}
+            host-matrix-output: ${{ steps.get-matrices.outputs.host-matrix-output }}
+        steps:
+            - uses: actions/checkout@v4
+            - id: get-matrices
+              uses: ./.github/workflows/create-tests-matrices
+              with:
+                  language-name: node
+            - name: Echo outputs
+              run: |
+                  echo "should_run: ${{ steps.get-matrices.outputs.should_run }}"
+                  echo "engine-matrix-output: ${{ steps.get-matrices.outputs.engine-matrix-output }}"
+                  echo "host-matrix-output: ${{ steps.get-matrices.outputs.host-matrix-output }}"
+    create-version-matrix:
+        runs-on: ubuntu-latest
+        needs: get-matrices
+        if: ${{ needs.get-matrices.outputs.should-run == 'true' }}
+        outputs:
+            node-matrix-output: ${{ steps.create-version-matrix.outputs.node-matrix-output }}
+        steps:
+            - name: Checkout
+              uses: actions/checkout@v4
```

Comment:
Same - we don't need checkout, becase we don't touch sources below

Reply:
same as above